### DISCUSSION TITLE
Balancing the flexibility of ComfyUI with everyday practicality for a user-centric interface

### DISCUSSION THREAD

**Discussion by cmp-nct** [comfyanonymous/ComfyUI#2149] [General]
Probably not the most popular idea:

I found ComfyUI to be a extremely well developed tool but it lacks that "refinement" for actual comfortable use. 
The speed of inovation, internal performance, flexibility .. that's all seriously great work. Just also cumbersome to use.
ComfyUI is more a refined visual development environment than a real UI.

There should be an additional feature that converts selected elements of a workflow into a multi-tab browser experience that basically works and looks like Automatic1111 type UI.
This would combine the best of both worlds: a highly dynamic and easily changeable backend and a true "use-focused" UI.

I guess a few changes or additions would be needed, such as better integrated ways to bypass groups (as in switching features on and off) and a way to project output and controls into a tab-like environment.

**Comment by camoody1** [comment]
I remember watching a Scott Detweiler video months ago when I first started playing with Comfy where he said something to the effect that Comfy was designed to be a backend for others to use and then a sort of clumsy UI was just laid over top of it for use in the short term. Unfortunately, it looks as if this UI is staying for the long term. It's definitely NOT user friendly to the layman.

**Reply by cmp-nct** [reply]
It's not really useable for professional use, you can somehow adapt it to singular use cases and it performs well but it's not what someone who works with image AI usually would expect.
But the backened (I include the current frontend into that) is great, it could allow to power a UI

**Comment by WeeBull** [comment]
Please stop with the "ComfyUI should be more like A1111" train. You can't praise the "speed of innovation, internal performance, flexibility" and then, in the same breath, say it should have something that "looks like Automatic1111 type UI". 

The things you praise are a product of it having the extremely modular design that it does, and a A1111 UI does not allow for that. Doing what you ask would kill the innovation that this platform enables. 

Maybe [Cushy Studio](https://github.com/rvion/CushyStudio) is more to your liking to get a mix of the two, but ComfyUI needs to stay broadly as it is. It's not for everyone, **and that's ok.**

**Reply by cmp-nct** [reply]
Not sure why a significant feature improvement that makes it great for everyone would be "not ok", you didn't explain that part?

**Reply by camoody1** [reply]
There are a lot of improvements that could be made to the current UI to improve the usability **dramatically** without affecting the power of the tool. It doesn't have to be ugly and hard to use in order to be powerful.

**Reply by ltdrdata** [reply]
Since the UI of ComfyUI started as a PoC, there are many flaws in various aspects. However, the focus should be on improving completeness in terms of node-based UX rather than steering towards a direction similar to A1111. 

The UX that conceals nodes and complex workflows, exposing only functionalities, is already being worked on in a much better direction by 3rd party front ends.

I recommend taking a look at this: https://github.com/rvion/CushyStudio

While it has taken time for the project to be publicly released, leading many to doubt its legitimacy as vaporware, direct communication with the developer confirms that work is indeed underway. It seems that the project is simply aiming for a high level of completion. [SDFX](https://www.reddit.com/r/comfyui/comments/16ncwlj/sdfx_studio_grade_new_comfy_ui_free_opensource/)

**Reply by rvion** [reply]
Hi there, cushy studio dev here. I would also recommend you to try it out :) 

Small note: CushyStudio is not related to sdfx at all. I'm in contact with them, but have a much much different plan for the future :)

**Reply by ltdrdata** [reply]
Oh, it seems like it was written in a way that could be misunderstood. SDFX is referring to a completely different front end that is currently in development. lol
